### ISSUE TITLE
Cannot login to Mi Account when running HA UI remotely

### ISSUE THREAD

**Issue by eschweighofer** [AlexxIT/XiaomiGateway3#979]
Hello,

I am not able to login to my MI Account from XiaomiGateway3 (running latest 3.1.0 in docker)
**Issue is ONLY when UI is opened outside the docker and public IP of docker and device where UI is opened don to match**

The password is accepted, because when I enter a wrong one, I get : Can't login, check username and password

After entering the correct one, I get the 'Click to verify URL', so I click on it, opens in another tab, get the verification code, enter it and the web returns 'OK'. so far so good. 

![image](https://user-images.githubusercontent.com/37148906/218313390-2d7c941e-71b6-43fd-8ad5-9e4f1aa706d9.png)

When I go to the previous tab, I still see the UI and it does not login... - When I click 'submit' again I get a new verification URL.... so I am stuck in circle...

Debug shown only
```
2023-02-12 14:14:25.241 DEBUG (MainThread) [custom_components.xiaomi_gateway3.core.xiaomi_cloud] MiCloud step1
2023-02-12 14:14:25.314 DEBUG (MainThread) [custom_components.xiaomi_gateway3.core.xiaomi_cloud] MiCloud step2
```

Any idea?
Thanks

**Comment by eschweighofer** [comment]
the devices are present in my Mi Account as seen from another tool.

![image](https://user-images.githubusercontent.com/37148906/218313714-08c67669-751e-4721-b728-2271f57b51e5.png)

**Comment by eschweighofer** [comment]
with some debugging I found the spot where it fails...

In the response from mi server I have empty location
![image](https://user-images.githubusercontent.com/37148906/218339141-bc61d992-d3da-48be-97d8-2220fcc8925c.png)

and in the code in this case we return False and do not continue to step3, where we should get the token from verification URL response

```
# xiaomi_cloud.py

class MiCloud:
    auth = None
    verify = None

    def __init__(self, session: ClientSession, servers: list = None):
        self.session = session
        self.servers = servers or ["cn"]
        self.device_id = get_random_string(16)

    async def login(self, username: str, password: str):
        try:
            payload = await self._login_step1()
            data = await self._login_step2(username, password, payload)
            self.verify = data.get("notificationUrl")
            if not data["location"]:
                return False

------


async def _login_step3(self, location):
        r = await self.session.get(location, headers={"User-Agent": UA})
        service_token = r.cookies["serviceToken"].value
        _LOGGER.debug(f"MiCloud step3")
        return service_token



```

**Comment by eschweighofer** [comment]
Ok, now I got the root cause, the public IP of the docker where HA is running and the device where HA UI is opened and integration is added must match. to achieve this I did the following:

1. setup a local socks proxy on my PC `ssh -D 8088 [username]@[host where HA is running or host where docker is running]`
2. execute chrome with following parameters to use that proxy `--proxy-server="socks5://localhost:8088" --host-resolver-rules="MAP * ~NOTFOUND , EXCLUDE localhost"

more info here https://dltj.org/article/ssh-as-socks-proxy/ and here https://www.chromium.org/developers/design-documents/network-stack/socks-proxy/
